MapboxNavigation/NavigationViewController.swift

```diff
@@ -300,49 +308,30 @@ open class NavigationViewController: UIViewController {
     /**
      If `true`, `UIApplication.isIdleTimerDisabled` is set to `true` in `viewWillAppear(_:)` and `false` in `viewWillDisappear(_:)`. If your application manages the idle timer itself, set this property to `false`.
      */
-    @objc public var shouldManageApplicationIdleTimer = true
+    public var shouldManageApplicationIdleTimer = true
     
     /**
      Bool which should be set to true if a CarPlayNavigationView is also being used.
      */
-    @objc public var isUsedInConjunctionWithCarPlayWindow = false {
+    public var isUsedInConjunctionWithCarPlayWindow = false {
         didSet {
-            self.mapViewController?.isUsedInConjunctionWithCarPlayWindow = self.isUsedInConjunctionWithCarPlayWindow
+            self.mapViewController.isUsedInConjunctionWithCarPlayWindow = self.isUsedInConjunctionWithCarPlayWindow
         }
     }
     
-    var isConnectedToCarPlay: Bool {
-        if #available(iOS 12.0, *) {
-            CarPlayManager.shared.isConnectedToCarPlay
-        } else {
-            false
-        }
-    }
-    
-    var mapViewController: RouteMapViewController?
-    
     /**
      A Boolean value that determines whether the map annotates the locations at which instructions are spoken for debugging purposes.
      */
-    @objc public var annotatesSpokenInstructions = false
-    
-    var styleManager: StyleManager!
-    
-    private var locationManager: NavigationLocationManager!
-    
-    var currentStatusBarStyle: UIStatusBarStyle = .default {
-        didSet {
-            self.mapViewController?.instructionsBannerView.backgroundColor = InstructionsBannerView.appearance().backgroundColor
-            self.mapViewController?.instructionsBannerContentView.backgroundColor = InstructionsBannerContentView.appearance().backgroundColor
-        }
-    }
-    
-    override open var preferredStatusBarStyle: UIStatusBarStyle {
-        self.currentStatusBarStyle
-    }
+    public var annotatesSpokenInstructions = false
     
+    // MARK: - Lifecycle
+	
     public required init?(coder aDecoder: NSCoder) {
+        self.directions = .shared
+        self.mapViewController = RouteMapViewController(routeController: self.routeController)
+        self.locationManager = NavigationLocationManager()
```

Comment:
Just so I understand: These assignments are a consequence of getting rid of the implicitly unwrapped optionals above, right?

(seems like a good tradeoff 👍)

Reply:
yes, the same is happening in the other init methods